widget/form.go

```diff
@@ -243,61 +242,48 @@ func (f *Form) ensureRenderItems() {
 
 func (f *Form) setUpValidation(widget fyne.CanvasObject, i int) {
 	updateValidation := func(err error) {
-		if err == errFormItemInitialState {
-			return
-		}
 		f.Items[i].validationError = err
 		f.Items[i].invalid = err != nil
+		if f.rendered {
+			f.Items[i].changedAfterRender = true
+		}
 		f.setValidationError(err)
 		f.checkValidation(err)
 		f.updateHelperText(f.Items[i])
 	}
 	if w, ok := widget.(fyne.Validatable); ok {
-		f.Items[i].invalid = w.Validate() != nil
-		if e, ok := w.(*Entry); ok {
-			e.onFocusChanged = func(bool) {
-				updateValidation(e.validationError)
-			}
-			if e.Validator != nil && f.Items[i].invalid {
-				// set initial state error to guarantee next error (if triggers) is always different
-				e.SetValidationError(errFormItemInitialState)
-			}
-		}
+		err := w.Validate()
+		f.Items[i].invalid = err != nil
+		f.Items[i].validationError = err
 		w.SetOnValidationChanged(updateValidation)
 	}
 }
 
 func (f *Form) setValidationError(err error) {
-	if err == nil && f.validationError == nil {
-		return
-	}
-
-	if !errors.Is(err, f.validationError) {
-		if err == nil {
-			for _, item := range f.Items {
-				if item.invalid {
-					err = item.validationError
-					break
+	if err == nil {
+		for _, item := range f.Items {
+			if item.invalid {
+				if item.validationError == nil {
+					item.validationError = errors.New("")
```

Comment:
Its necessary for this check here: https://github.com/fyne-io/fyne/pull/2978/files#diff-dd8273e503eced03ee09025bf7098f8dfdf8e73092e9e4c90fdc89aa2154d456R286 

Although considering that I've added [this](https://github.com/fyne-io/fyne/pull/2978/files#diff-dd8273e503eced03ee09025bf7098f8dfdf8e73092e9e4c90fdc89aa2154d456R257) it might not be necessary now that I look at it again.